pkg/vendir/directory/staging_dir.go

```diff
@@ -85,7 +86,17 @@ func (d StagingDir) TempArea() StagingTempArea {
 }
 
 func (d StagingDir) CleanUp() error {
-	return d.cleanUpAll()
+	// workaround where os.RemoveAll would not delete folder
+	var err error
+	for {
+		stat, e := os.Stat(d.rootDir)
```

Comment:
The case exists where the stat fails for a reason other than a NotExist error. In this case, I would expect the method to return an error since cleanup failed, but in this implementation, it will return `nil`.

Is there any more info about the underlying reason that os.RemoveAll wasn't cleaning up the directory? If it's possible, I think getting to the bottom of that is more desirable than implementing a retry.

Reply:
I don't know what is causing the folder not being removed, I did try a few things prior to adding this hack.

- I disabled the AV as this would be the first suspect
- I added some delay at the end of the program to see if maybe the rm was done in the background and the prog exited to fast

the folders were still not deleted.

I will add a check on the error type as you suggested.